### ISSUE TITLE
Responsive Collapse

### ISSUE THREAD

**Issue by nhcodes** [twbs/bootstrap#38752]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

It would be very helpful to have responsive classes for the collapse component.  For example adding a .collapse-sm class would mean that the content is hidden by default only if the viewport is smaller than 576px.

Example:
![output](https://github.com/twbs/bootstrap/assets/120429276/ab18916f-727c-4086-8b61-d63cb08856f6)

Example Code: https://jsfiddle.net/mvLxytk7/1/

### Motivation and context

It's already possible to achieve this by using the .navbar-collapse, and .navbar-expand{-sm|-md|-lg|-xl|-xxl} classes like in [this example](https://getbootstrap.com/docs/5.3/components/navbar/#responsive-behaviors). But I think it would be easier and make more sense if one would be able to accomplish this without having to use a navbar.

**Comment by michaelhiiva** [comment]
I agree with this. It would be very nice to have this as a feature of collapse. Say for example if one were using this on a menu as in the offcanvas sidebar (see; https://getbootstrap.com/docs/5.2/examples/sidebars/#) , but one only wanted to have a collapse menu on smaller but not larger screens. I was working on a project: https://github.com/michaelhiiva/Introduction-with-Section-Dropdown-Frontend-Mentor, where one needs a dropdown on larger screens and collapse menu on smaller screens. Right now, the only way I was able to achieve this was with JavaScript. It would have been nice to just use the standards or norms already in place within Bootstrap.

**Comment by Arvind-Suthar** [comment]
I have worked on a similar issue during a project of my own. I would like to make a pull request for this with my solution

**Comment by mdo** [comment]
Skipping for now.
